summingbird-batch/src/main/scala/com/twitter/summingbird/batch/BatchID.scala

```diff
@@ -93,7 +93,7 @@ object BatchID {
       .flatMap {
         case (min, max, cnt) =>
           if ((min + cnt) == (max + 1L)) {
-            Some(Interval.leftClosedRightOpen(min, max.next).right.get)
+            Some(Intersection(InclusiveLower(min), InclusiveUpper(max)))
```

Comment:
code *may* be assuming this interval is left closed right open.

Maybe it is better to check if `max ==BatchID(Long.MaxValue)` and only in that case do the InclusiveUpper?